torch/csrc/jit/runtime/symbolic_script.cpp

```diff
@@ -222,6 +222,17 @@ const std::vector<std::string> functions = {
             # FIXME: torchscript: torch.zeros(sizes, grad.options())
             return torch.zeros(sizes).to(grad).scatter_(dim, indices, grad)
 
+        def gather(self,
+                    dim: int,
+                    index,
+                    *,
+                    sparse_grad: bool = False):
+            output = torch.gather(self, dim, index)
+            def backward(grad_output):
+                grad_self = torch.zeros_like(self).scatter_add_(dim, index, grad_output)
```

Comment:
nitpick: we are dropping `sparse_grad` on backwards, which doesn't seem to be right. Let's at least add a conditional here and go through `torch.gather_backward` when `sparse_grad` is True.

I'm hoping that `sparse_grad` will be constant most times and the conditional block can be constant folded, so we'll still be able to fuse the backward ops.

Reply:
Ok, but I don't know how to call `torch.gather_backward` here. When `sparse_grad` is True, the actually executed function is `at::_gather_sparse_backward`, but it does not provide python interface (maybe just my mistake). Maybe I need reimplement it in python?

Reply:
Sorry I didn't respond in time for this. I see you already did your python version here.

Does this not work? vvv
`
if (sparse_grad):
    return torch.gather_backward(grad_output, self, dim, True), None, None, None
`

Just FYI:
1. You can check out aten signature here: https://github.com/pytorch/pytorch/blob/dabf515c184c18eb5a7ec2562110f957eb899427/aten/src/ATen/native/native_functions.yaml#L8556-L8559
2. autograd rule can be looked up here: https://github.com/pytorch/pytorch/blob/dabf515c184c18eb5a7ec2562110f957eb899427/tools/autograd/derivatives.yaml#L687-L690

Reply:
I think `gather_backward` is not registered in torch, it just an `aten` function.